samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/mesh/models.h>
+#include <dk_buttons_and_leds.h>
+#include "model_handler.h"
+
+#define TEMP_RANGE_COLUMNS_THRESHOLD 32
+
+/* The columns (temperature ranges) for relative
+ * runtime in a chip temperature
+ */
+static const struct bt_mesh_sensor_column columns[] = {
+	{ { 0 }, { TEMP_RANGE_COLUMNS_THRESHOLD } },
+	{ { TEMP_RANGE_COLUMNS_THRESHOLD }, { 100 } },
+};
+
+static struct device *dev;
+static u32_t tot_temp_samps;
+static u32_t col_samps[ARRAY_SIZE(columns)];
+static struct sensor_value relative_runtime[ARRAY_SIZE(columns)];
+
+static s32_t prev_pres;
+
+static int chip_temp_get(struct bt_mesh_sensor *sensor,
+			 struct bt_mesh_msg_ctx *ctx, struct sensor_value *rsp)
+{
+	int err;
+
+	sensor_sample_fetch(dev);
+
+	err = sensor_channel_get(dev, SENSOR_CHAN_DIE_TEMP, rsp);
+
+	if (rsp->val1 < TEMP_RANGE_COLUMNS_THRESHOLD) {
+		col_samps[0]++;
+	} else {
+		col_samps[1]++;
+	}
+
+	tot_temp_samps++;
+
+	return err;
+}
+
+static struct bt_mesh_sensor chip_temp = {
+	.type = &bt_mesh_sensor_present_dev_op_temp,
+	.get = chip_temp_get,
+};
+
+static int relative_runtime_in_chip_temp_get(
+	struct bt_mesh_sensor *sensor, struct bt_mesh_msg_ctx *ctx,
+	const struct bt_mesh_sensor_column *column, struct sensor_value *value)
+{
+	s32_t index = column - &columns[0];
+
+	if (col_samps[index] > 0) {
+		u8_t percent_steps = (200 * col_samps[index]) / tot_temp_samps;
+
+		relative_runtime[index].val1 = percent_steps / 2;
+		relative_runtime[index].val2 = (percent_steps % 2) * 500000;
```

Comment:
No need to make relative_runtime a global variable, it is overwritten before it's used every time anyway. Also no need for the if. You'll never divide by zero here anyway, and simplicity is preferred over performance in samples. Just do

```C
u8_t percent_steps = (200 * col_samps[index]) / tot_temp_samps;

value[0].val1 = percent_steps / 2;
value[0].val2 = (percent_steps % 2) * 500000;
```

Reply:
Done